imap_processing/lo/l1a/lo_l1a.py

```diff
@@ -29,156 +29,139 @@ def lo_l1a(dependency: Path, data_version: str) -> list[Path]:
     created_file_paths : list[Path]
         Location of created CDF files.
     """
-    # TODO: decom the CCSDS file
-
-    # TODO: sort the decommed packets by SHCOARSE
-
-    # TODO: group the packets by APID
-
-    # TODO: Loop through packets
-    # TODO: Inside Loop: Check for each APID and Initialize data
-    #  class object with packet contents, add to list
+    xtce_file = imap_module_directory / "lo/packet_definitions/lo_xtce.xml"
+    datasets_by_apid = packet_file_to_datasets(
+        packet_file=dependency.resolve(),
+        xtce_packet_definition=xtce_file.resolve(),
+    )
 
     # create the attribute manager for this data level
     attr_mgr = ImapCdfAttributes()
     attr_mgr.add_instrument_global_attrs(instrument="lo")
     attr_mgr.add_instrument_variable_attrs(instrument="lo", level="l1a")
     attr_mgr.add_global_attribute("Data_version", data_version)
 
-    # TODO: replace with real processing when sample data is available
-    # Temporary code until I get sample data. The fields will still be pulled
-    # from the data classes, but the data class will be determined by the
-    # APID in the decommed file and the data class will first be populated
-    # with the packet data.
-    packet_file_name = dependency.name
-    if "_de_" in packet_file_name:
-        data_fields = fields(ScienceDirectEvents)
+    if LoAPID.ILO_SCI_CNT in datasets_by_apid:
+        logical_source = "imap_lo_l1a_histogram"
+        datasets_by_apid[LoAPID.ILO_SCI_CNT] = parse_histogram(
+            datasets_by_apid[LoAPID.ILO_SCI_CNT], attr_mgr
+        )
+        datasets_by_apid[LoAPID.ILO_SCI_CNT] = add_dataset_attrs(
+            datasets_by_apid[LoAPID.ILO_SCI_CNT], attr_mgr, logical_source
+        )
+    # TEMPORARY: datasets are being removed until the logic
+    # is written to add the attributes in a future PR. I only
+    # have one sample dataset containing all the packets so I
+    # cannot test these easily in isolation
```

Comment:
Rather than deleting them, what about only indexing the return back out of this function with the ones you can test?

```
good_apids = [LoAPID.ILO_SCI_CNT]  # Add to this as you go
return [datasets_by_apid[good_apid] for good_apid in good_apids])
```